module/epochs/epoch_config.go

```diff
@@ -0,0 +1,150 @@
+	CurrentEpochCounter          cadence.UInt64
+	NumViewsInEpoch              cadence.UInt64
+	NumViewsInStakingAuction     cadence.UInt64
+	NumViewsInDKGPhase           cadence.UInt64
+	NumCollectorClusters         cadence.UInt16
+	FLOWsupplyIncreasePercentage cadence.UFix64
+	RandomSource                 cadence.String
+	CollectorClusters            flow.AssignmentList
+	ClusterQCs                   []*flow.QuorumCertificate
+	DKGPubKeys                   []crypto.PublicKey
+}
+
+// DefaultEpochConfig returns an EpochConfig with default values used for
+// testing.
+func DefaultEpochConfig() EpochConfig {
+	return EpochConfig{
+		CurrentEpochCounter:          cadence.UInt64(0),
+		NumViewsInEpoch:              cadence.UInt64(100),
+		NumViewsInStakingAuction:     cadence.UInt64(10),
+		NumViewsInDKGPhase:           cadence.UInt64(10),
+		NumCollectorClusters:         cadence.UInt16(3),
+		FLOWsupplyIncreasePercentage: cadence.UFix64(5),
+	}
+}
+
+// EncodeClusterAssignment encodes an AssigmentList into a byte array that can
+// be used as a transaction argument when deploying the epochs contract.
+func EncodeClusterAssignments(clusterAssignments flow.AssignmentList, service flow.Address) []byte {
+	collectorClusterValues := []cadence.Value{}
+
+	for i, cluster := range clusterAssignments {
+		clusterIndex := cadence.UInt16(i)
+
+		weightsByNodeID := []cadence.KeyValuePair{}
+		for _, id := range cluster {
+			kvp := cadence.KeyValuePair{
+				Key:   cadence.NewString(id.String()),
+				Value: cadence.NewUInt64(0),
+			}
+			weightsByNodeID = append(weightsByNodeID, kvp)
+		}
+
+		fields := []cadence.Value{
+			clusterIndex,
+			cadence.NewDictionary(weightsByNodeID),
+		}
+
+		clusterStruct := cadence.NewStruct(fields).
+			WithType(&cadence.StructType{
+				Location: common.AddressLocation{
+					Address: common.BytesToAddress(service.Bytes()),
+					Name:    "Service",
+				},
+				QualifiedIdentifier: "FlowEpochClusterQC.Cluster",
+				Fields: []cadence.Field{
+					{
+						Identifier: "index",
+						Type:       cadence.UInt16Type{},
+					},
+					{
+						Identifier: "nodeWeights",
+						Type: cadence.DictionaryType{
+							KeyType:     cadence.StringType{},
+							ElementType: cadence.UInt64Type{},
+						},
+					},
+				},
+			})
+
+		collectorClusterValues = append(collectorClusterValues, clusterStruct)
+	}
+
+	collectorClusters := cadence.NewArray(collectorClusterValues)
+
+	return jsoncdc.MustEncode(collectorClusters)
+}
+
+// EncodeClusterQCs encodes a slice of QuorumCertificates into a byte array that
+// can be used as a transaction argument when deploying the epochs contract.
+func EncodeClusterQCs(qcs []*flow.QuorumCertificate, service flow.Address) []byte {
+	qcValues := []cadence.Value{}
+
+	for i, qc := range qcs {
+		qcIndex := cadence.UInt16(i)
+
+		qcVotes := []cadence.Value{}
+		for _, v := range qc.SignerIDs {
+			qcVotes = append(qcVotes, cadence.NewString(v.String()))
+		}
```

Comment:
Here we are adding signer IDs rather than votes. I don't think it will matter and it looks like recovering the actual individual votes would be a bit tricky, but we should add a comment noting this and why it won't matter (these initial values aren't used by the contract).